virtual_rainforest/models/soil/carbon.py

```diff
@@ -1,34 +1,60 @@
 """The ``models.soil.carbon`` module  simulates the soil carbon cycle for the Virtual
-Rainforest. At the moment four pools are modelled, these are low molecular weight carbon
-(LMWC), mineral associated organic matter (MAOM), microbial biomass, and particulate
-organic matter (POM).
+Rainforest. At the moment five pools are modelled, these are low molecular weight carbon
+(LMWC), mineral associated organic matter (MAOM), microbial biomass, particulate organic
+matter (POM), and POM degrading enzymes.
 """  # noqa: D205, D415
 
+from dataclasses import dataclass
+
 import numpy as np
 from numpy.typing import NDArray
-from scipy.constants import convert_temperature, gas_constant
 
-from virtual_rainforest.core.logger import LOGGER
+from virtual_rainforest.core.constants import CoreConsts
 from virtual_rainforest.models.soil.constants import SoilConsts
+from virtual_rainforest.models.soil.env_factors import (
+    calculate_environmental_effect_factors,
+    calculate_leaching_rate,
+    calculate_temperature_effect_on_microbes,
+)
 
-# TODO - Once enzymes are added, temperature dependence of saturation constants should
-# be added.
 
+@dataclass
+class MicrobialBiomassLoss:
+    """A data class to store the various biomass losses from microbial biomass."""
 
+    maintenance_synthesis: NDArray[np.float32]
+    """Rate at which biomass must be synthesised to balance losses [kg C m^-3 day^-1].
+    """
+    pom_enzyme_production: NDArray[np.float32]
+    """Rate at which POM degrading enzymes are produced [kg C m^-3 day^-1]."""
+    maom_enzyme_production: NDArray[np.float32]
+    """Rate at which MAOM degrading enzymes are produced [kg C m^-3 day^-1]."""
+    necromass_decay_to_lmwc: NDArray[np.float32]
+    """Rate at which biomass is lost to the LMWC pool [kg C m^-3 day^-1]."""
+    necromass_decay_to_pom: NDArray[np.float32]
+    """Rate at which biomass is lost to the POM pool [kg C m^-3 day^-1]."""
+
+
+# TODO - This function should probably be shortened, leaving as is for the moment as a
+# sensible split will probably be more obvious once more is added to this function.
```

Comment:
This function is indeed rather a beast. It is fine to leave for now (though maybe open an issue to fix it later?), but there's an awful lot going on and the number of arguments is also too high.

I don't really understand enough to give a better design for this, but in general I would say:

1. If lots of the arguments to this function are just being passed through to one or two other functions, you could just pass in the *results* of these functions instead and have the caller calculate them
2. If some of the arguments would naturally group together, consider bundling them together in a class/dataclass (provided that this function won't be the only user of it)

Reply:
Oh yes, I should make an issue!

Reply:
Have made issue #344 for this problem